frontend/src/app/shared/components/storages/storage/storage.component.ts

```diff
@@ -300,6 +312,58 @@ export class StorageComponent extends UntilDestroyedMixin implements OnInit, OnD
       );
   }
 
+  private uploadAndNotify(link:IUploadLink, file:UploadFile):Observable<IFileLinkOriginData> {
+    const { method, href } = link._links.destination;
+
+    interface FileUploadResponse {
+      file_name:string;
+      file_id:string;
+    }
+
+    const formData = new FormData();
+    formData.append('file', file, file.name);
+    const observable = this.http.request<FileUploadResponse>(
+      method,
+      href,
+      {
+        body: formData,
+        headers: { 'X-External-Request': 'true' },
+        observe: 'events',
+        reportProgress: true,
+        responseType: 'json',
+      },
+    ).pipe(share());
+    const notification = this.toastService.add({
+      data: [[file, observable]],
+      type: 'upload',
+      message: this.text.toast.uploadingLabel,
+    });
+
+    return observable
+      .pipe(
+        tap(() => {
+          setTimeout(() => this.toastService.remove(notification), 700);
```

Comment:
Not for this PR, but wouldn't it be nice if the close timeout could be configured while creating the toast and kept track in the `ToastService`? And maybe not with `setTimeout` nut with `pipe(delay(...))`? This side effect here bothers me.

Reply:
yes, sounds sensible, yet the refactoring of the toast service would exceed the current scope of this WP.